frontend/src/App.svelte

```diff
@@ -3,12 +3,17 @@
   import { Link } from 'carbon-components-svelte'
   import { routes } from './routes'
   import LanguageSelector from './lib/LanguageSelector.svelte'
+  import { setLanguage } from './lib/stores'
+
+  function routeLoading(event) {
+    setLanguage(event.detail.params.lang)
```

Comment:
And this should be `$:` or similar on the `Main` component monitoring the `params` for changes.